transcendence-app/src/user/user.controller.ts

```diff
@@ -26,6 +27,27 @@ import {
   UsersPaginationQueryDto,
 } from './dto/user.pagination.dto';
 import { User as GetUser } from './decorators/user.decorator';
+import LocalFileInterceptor from '../shared/local-file/local-file.interceptor';
+import { AVATARS_PATH } from './constants';
+
+export const AvatarFileInterceptor = LocalFileInterceptor({
+  fieldName: 'file',
+  path: AVATARS_PATH,
+  fileFilter: (request, file, callback) => {
+    if (file.mimetype !== 'image/jpeg') {
+      return callback(
+        new UnprocessableEntityException(
+          'Validation failed (expected type is jpeg)',
+        ),
+        false,
+      );
+    }
+    callback(null, true);
+  },
+  limits: {
+    fileSize: Math.pow(1024, 2), // 1MB
+  },
+});
```

Comment:
This is an instance of the `LocalFileInterceptor`, which allows only JPEG files under 1MB.

We save this into a variable to allow importing it from the unit tests

Reply:
Why only jpg? Are we against any other image format? Anyway, we can discuss this later.

Reply:
We could allow more image formats :+1:

Reply:
Update: I've added a constant with an array of allowed formats. For now it contains `image/jpeg`, `image/png` and `image/webp`. This way, we can easily add/remove other image formats in the future.